optuna_dashboard/_preferential_history.py

```diff
@@ -43,38 +43,50 @@ def report_history(
     study_id: int,
     storage: BaseStorage,
     input_data: NewHistory,
-) -> None:
+) -> str:
     preferences = []
     # TODO(moririn): Use TypeGuard after adding other history types.
     if input_data.mode == "ChooseWorst":
         preferences = [
-            (best, input_data.clicked)
-            for best in input_data.candidates
-            if best != input_data.clicked
+            (better, input_data.clicked)
+            for better in input_data.candidates
+            if better != input_data.clicked
         ]
     else:
         assert False, f"Unknown data: {input_data}"
 
-    preference_id = report_preferences(
+    id = report_preferences(
         study_id=study_id,
         storage=storage,
         preferences=preferences,
     )
-    history_id = str(uuid.uuid4())
 
     if input_data.mode == "ChooseWorst":
         history: ChooseWorstHistory = {
             "mode": "ChooseWorst",
-            "id": history_id,
-            "preference_id": preference_id,
+            "id": id,
             "timestamp": datetime.now().isoformat(),
             "candidates": input_data.candidates,
             "clicked": input_data.clicked,
+            "preferences": preferences,
         }
 
-    key = _SYSTEM_ATTR_PREFIX_HISTORY + history_id
+    key = _SYSTEM_ATTR_PREFIX_HISTORY + id
     storage.set_study_system_attr(
         study_id=study_id,
         key=key,
         value=json.dumps(history),
     )
+    return id
+
+
+def remove_history(study_id: int, storage: BaseStorage, id: str) -> None:
```

Comment:
You should not use `id` as a variable name since it conflicts with the built-in function `id()`.

```suggestion
def remove_history(study_id: int, storage: BaseStorage, history_id: str) -> None:
```